riscv/csrs.cc

```diff
@@ -516,6 +516,24 @@ reg_t mstatus_csr_t::compute_mstatus_initial_value() const noexcept {
          | 0;  // initial value for mstatus
 }
 
+// implement class mnstatus_csr_t
+mnstatus_csr_t::mnstatus_csr_t(processor_t* const proc, const reg_t addr):
+  basic_csr_t(proc, addr, 0) {
+}
+
+bool mnstatus_csr_t::unlogged_write(const reg_t val) noexcept {
+  // NMIE can be set but not cleared
+  const reg_t mask = (~read() & MNSTATUS_NMIE)
+                   | (proc->extension_enabled('H') ? MNSTATUS_MNPV : 0)
```

Comment:
I haven't read the spec but I'm guessing this MNPV should be 0 if no hypervisor? What if we write it to 1, and then disable H?

Most bits that are H-only are masked on read, not write, for this reason. Or written to 0 when H is cleared in `misa_csr_t::unlogged_write()`.

Reply:
I'm following our implementation of `mstatus.MPV`.  Is the implication that needs to change, too?